python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -90,37 +114,42 @@ def cuda_array_interface_wrapper(
     )
 
 
-class Buffer(Serializable):
-    """A Buffer represents device memory.
+class BufferOwner(Serializable):
+    """A owning buffer that represents device memory.
 
-    Use the factory function `as_buffer` to create a Buffer instance.
+    This class isn't meant to be used throughout cuDF. Instead, it
+    standardizes data owning by wrapping any data object that
+    represents device memory. Multiple `Buffer` instances, which are
+    the ones used throughout cuDF, can then refer to the same
+    `BufferOwner` instance.
+
+    Use `_from_device_memory` and `_from_host_memory` to create
+    a new instance from either device or host memory respectively.
     """
 
     _ptr: int
     _size: int
     _owner: object
 
-    def __init__(self):
-        raise ValueError(
-            f"do not create a {self.__class__} directly, please "
-            "use the factory function `cudf.core.buffer.as_buffer`"
-        )
```

Comment:
**question:** We've now lost this "don't call init" path. Is that deliberate?

Reply:
Fixed